File: crates/protocol-conformance/src/generated.rs
// 本文件由 tools/protocol-codegen.ts 生成，禁止手工修改。

use std::collections::BTreeMap;

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ActorRef {
    pub agent: AgentRef,
    pub instance_id: String,
    pub provenance: Provenance,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentRef {
    pub agent_id: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub avatar_url: Option<String>,
    pub display_name: String,
    pub matrix_user_id: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentStatusEvent {
    pub actor: ActorRef,
    pub correlation_id: String,
    pub created_at: String,
    pub event_type: String,
    pub id: String,
    pub lease_expires_at: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub progress: Option<f64>,
    pub schema_version: String,
    pub signature: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub started_at: Option<String>,
    pub status: AgentWorkStatus,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub task_summary: Option<String>,
    pub visibility: AgentStatusVisibility,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum AgentStatusVisibility {
    #[serde(rename = "coarse")]
    Coarse,
    #[serde(rename = "detailed")]
    Detailed,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum AgentWorkStatus {
    #[serde(rename = "offline")]
    Offline,
    #[serde(rename = "idle")]
    Idle,
    #[serde(rename = "working")]
    Working,
    #[serde(rename = "waiting_input")]
    WaitingInput,
    #[serde(rename = "blocked")]
    Blocked,
    #[serde(rename = "completed")]
    Completed,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CapabilityManifest {
    pub event_types: Vec<String>,
    pub features: Vec<String>,
    pub protocol_versions: Vec<String>,
    pub schema_version: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ClientContentEncryption {
    pub algorithm: String,
    pub context_id: String,
    pub key_base64_url: String,
    pub nonce_base64_url: String,
    pub plaintext_size_bytes: u64,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ContentRef {
    pub content_id: String,
    pub digest_sha256: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub encryption: Option<ClientContentEncryption>,
    pub fetch_mode: String,
    pub media_type: String,
    pub size_bytes: u64,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum ErrorCategory {
    #[serde(rename = "validation")]
    Validation,
    #[serde(rename = "authentication")]
    Authentication,
    #[serde(rename = "authorization")]
    Authorization,
    #[serde(rename = "conflict")]
    Conflict,
    #[serde(rename = "transient")]
    Transient,
    #[serde(rename = "unknown_commit")]
    UnknownCommit,
    #[serde(rename = "dependency_unavailable")]
    DependencyUnavailable,
    #[serde(rename = "incompatible_version")]
    IncompatibleVersion,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ErrorEnvelope {
    pub category: ErrorCategory,
    pub code: String,
    pub correlation_id: String,
    pub details: BTreeMap<String, serde_json::Value>,
    pub message: String,
    pub retryable: bool,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub retry_after_seconds: Option<u64>,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum HandoffPermission {
    #[serde(rename = "read_text")]
    ReadText,
    #[serde(rename = "read_attachments")]
    ReadAttachments,
    #[serde(rename = "include_metadata")]
    IncludeMetadata,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum HandoffPurpose {
    #[serde(rename = "inspect")]
    Inspect,
    #[serde(rename = "summarize")]
    Summarize,
    #[serde(rename = "reply_draft")]
    ReplyDraft,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HandoffReceiptEvent {
    pub actor: ActorRef,
    pub correlation_id: String,
    pub created_at: String,
    pub event_type: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub failure_code: Option<String>,
    pub id: String,
    pub requester_instance_id: String,
    pub schema_version: String,
    pub signature: String,
    pub status: HandoffReceiptStatus,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum HandoffReceiptStatus {
    #[serde(rename = "delivered")]
    Delivered,
    #[serde(rename = "consumed")]
    Consumed,
    #[serde(rename = "declined")]
    Declined,
    #[serde(rename = "revoked")]
    Revoked,
    #[serde(rename = "expired")]
    Expired,
    #[serde(rename = "failed")]
    Failed,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HandoffRequestEvent {
    pub actor: ActorRef,
    pub approved_at: String,
    pub approved_by_principal_id: String,
    pub content: ContentRef,
    pub correlation_id: String,
    pub created_at: String,
    pub event_type: String,
    pub expires_at: String,
    pub id: String,
    pub permissions: Vec<HandoffPermission>,
    pub purpose: HandoffPurpose,
    pub risk_flags: Vec<String>,
    pub schema_version: String,
    pub signature: String,
    pub source: HandoffSource,
    pub target_agent_id: String,
    pub target_instance_id: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HandoffSource {
    pub actor: ActorRef,
    pub event_id: String,
    pub message_id: String,
    pub room_id: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MessagePreview {
    pub content_type: String,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub language: Option<String>,
    pub risk_flags: Vec<String>,
    pub sensitivity: MessageSensitivity,
    pub summary: String,
    pub title: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MessagePreviewEvent {
    pub actor: ActorRef,
    pub content: ContentRef,
    pub correlation_id: String,
    pub created_at: String,
    pub event_type: String,
    pub id: String,
    pub preview: MessagePreview,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub relation: Option<MessageRelation>,
    pub room_id: String,
    pub schema_version: String,
    pub signature: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MessageRelation {
    pub kind: String,
    pub target_message_id: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MessageRevisionEvent {
    pub actor: ActorRef,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub content: Option<ContentRef>,
    pub correlation_id: String,
    pub created_at: String,
    pub event_type: String,
    pub id: String,
    pub kind: MessageRevisionKind,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub preview: Option<MessagePreview>,
    pub room_id: String,
    pub schema_version: String,
    pub signature: String,
    pub target_message_id: String,
    #[serde(default, flatten, skip_serializing_if = "BTreeMap::is_empty")]
    pub extensions: BTreeMap<String, serde_json::Value>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum MessageRevisionKind {
    #[serde(rename = "replace")]
    Replace,
    #[serde(rename = "redact")]
    Redact,
    #[serde(rename = "moderate")]
    Moderate,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum MessageSensitivity {
    #[serde(rename = "normal")]
    Normal,
    #[serde(rename = "sensitive")]
    Sensitive,
    #[serde(rename = "restricted")]
    Restricted,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub enum Provenance {
    #[serde(rename = "human")]
    Human,
    #[serde(rename = "human_confirmed_agent")]
    HumanConfirmedAgent,
    #[serde(rename = "autonomous_agent")]
    AutonomousAgent,
}

File: packages/protocol-types/src/generated.ts
// 本文件由 tools/protocol-codegen.ts 生成，禁止手工修改。

export type Actor = HumanActor | AgentActor;

export type ActorRef = {
  readonly agent: AgentRef;
  readonly instanceId: string;
  readonly provenance: Provenance;
} & Readonly<Record<string, unknown>>;

export type AgentActor = {
  readonly agent: AgentRef;
  readonly instanceId: string;
  readonly kind: "agent";
  readonly provenance: AgentProvenance;
} & Readonly<Record<string, unknown>>;

export type AgentProvenance = "human_confirmed_agent" | "autonomous_agent";

export type AgentRef = {
  readonly agentId: string;
  readonly avatarUrl?: string;
  readonly displayName: string;
  readonly matrixUserId: string;
} & Readonly<Record<string, unknown>>;

export type AgentStatusEvent = {
  readonly actor: ActorRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.agent.status.v1";
  readonly id: string;
  readonly leaseExpiresAt: string;
  readonly progress?: number;
  readonly schemaVersion: "1.0";
  readonly signature: string;
  readonly startedAt?: string;
  readonly status: AgentWorkStatus;
  readonly taskSummary?: string;
  readonly visibility: AgentStatusVisibility;
} & Readonly<Record<string, unknown>>;

export type AgentStatusVisibility = "coarse" | "detailed";

export type AgentWorkStatus = "offline" | "idle" | "working" | "waiting_input" | "blocked" | "completed";

export type CapabilityManifest = {
  readonly eventTypes: ReadonlyArray<string>;
  readonly features: ReadonlyArray<string>;
  readonly protocolVersions: ReadonlyArray<string>;
  readonly schemaVersion: "1.0";
} & Readonly<Record<string, unknown>>;

export type ClientContentEncryption = {
  readonly algorithm: "io.github.rainyflash.agentroom.content.aes-256-gcm.v1";
  readonly contextId: string;
  readonly keyBase64Url: string;
  readonly nonceBase64Url: string;
  readonly plaintextSizeBytes: number;
};

export type ContentRef = {
  readonly contentId: string;
  readonly digestSha256: string;
  readonly encryption?: ClientContentEncryption;
  readonly fetchMode: "on_demand";
  readonly mediaType: string;
  readonly sizeBytes: number;
} & Readonly<Record<string, unknown>>;

export type ConversationMessage = {
  readonly mentions: ReadonlyArray<string>;
  readonly text: string;
};

export type ErrorCategory = "validation" | "authentication" | "authorization" | "conflict" | "transient" | "unknown_commit" | "dependency_unavailable" | "incompatible_version";

export type ErrorEnvelope = {
  readonly category: ErrorCategory;
  readonly code: string;
  readonly correlationId: string;
  readonly details: Readonly<Record<string, unknown>>;
  readonly message: string;
  readonly retryable: boolean;
  readonly retryAfterSeconds?: number;
} & Readonly<Record<string, unknown>>;

export type HandoffPermission = "read_text" | "read_attachments" | "include_metadata";

export type HandoffPurpose = "inspect" | "summarize" | "reply_draft";

export type HandoffReceiptEvent = {
  readonly actor: ActorRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.handoff.receipt.v1";
  readonly failureCode?: string;
  readonly id: string;
  readonly requesterInstanceId: string;
  readonly schemaVersion: "1.0";
  readonly signature: string;
  readonly status: HandoffReceiptStatus;
} & Readonly<Record<string, unknown>>;

export type HandoffReceiptStatus = "delivered" | "consumed" | "declined" | "revoked" | "expired" | "failed";

export type HandoffRequestEvent = {
  readonly actor: ActorRef;
  readonly approvedAt: string;
  readonly approvedByPrincipalId: string;
  readonly content: ContentRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.handoff.request.v1";
  readonly expiresAt: string;
  readonly id: string;
  readonly permissions: ReadonlyArray<HandoffPermission>;
  readonly purpose: HandoffPurpose;
  readonly riskFlags: ReadonlyArray<string>;
  readonly schemaVersion: "1.0";
  readonly signature: string;
  readonly source: HandoffSource;
  readonly targetAgentId: string;
  readonly targetInstanceId: string;
} & Readonly<Record<string, unknown>>;

export type HandoffSource = {
  readonly actor: ActorRef;
  readonly eventId: string;
  readonly messageId: string;
  readonly roomId: string;
} & Readonly<Record<string, unknown>>;

export type HumanActor = {
  readonly avatarUrl?: string;
  readonly displayName: string;
  readonly kind: "human";
  readonly matrixUserId: string;
  readonly principalId: string;
} & Readonly<Record<string, unknown>>;

export type MessagePreview = {
  readonly contentType: string;
  readonly conversation?: ConversationMessage;
  readonly language?: string;
  readonly riskFlags: ReadonlyArray<string>;
  readonly sensitivity: MessageSensitivity;
  readonly summary: string;
  readonly title: string;
} & Readonly<Record<string, unknown>>;

export type MessagePreviewEvent = {
  readonly actor: ActorRef;
  readonly content: ContentRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.message.preview.v1";
  readonly id: string;
  readonly preview: MessagePreview;
  readonly relation?: MessageRelation;
  readonly roomId: string;
  readonly schemaVersion: "1.0";
  readonly signature: string;
} & Readonly<Record<string, unknown>>;

export type MessagePreviewEventV2 = {
  readonly actor: Actor;
  readonly content: ContentRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.message.preview.v2";
  readonly id: string;
  readonly preview: MessagePreview;
  readonly relation?: MessageRelation;
  readonly roomId: string;
  readonly schemaVersion: "2.0";
  readonly signature?: string;
} & Readonly<Record<string, unknown>>;

export type MessageRelation = {
  readonly kind: "reply";
  readonly targetMessageId: string;
} & Readonly<Record<string, unknown>>;

export type MessageRevisionEvent = {
  readonly actor: ActorRef;
  readonly content?: ContentRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.message.revision.v1";
  readonly id: string;
  readonly kind: MessageRevisionKind;
  readonly preview?: MessagePreview;
  readonly roomId: string;
  readonly schemaVersion: "1.0";
  readonly signature: string;
  readonly targetMessageId: string;
} & Readonly<Record<string, unknown>>;

export type MessageRevisionEventV2 = {
  readonly actor: Actor;
  readonly content?: ContentRef;
  readonly correlationId: string;
  readonly createdAt: string;
  readonly eventType: "io.github.rainyflash.agentroom.message.revision.v2";
  readonly id: string;
  readonly kind: MessageRevisionKind;
  readonly preview?: MessagePreview;
  readonly roomId: string;
  readonly schemaVersion: "2.0";
  readonly signature?: string;
  readonly targetMessageId: string;
} & Readonly<Record<string, unknown>>;

export type MessageRevisionKind = "replace" | "redact" | "moderate";

export type MessageSensitivity = "normal" | "sensitive" | "restricted";

export type ModerationNoticeEvent = {
  readonly actionId: string;
  readonly eventType: "io.github.rainyflash.agentroom.moderation.notice.v1";
  readonly hidden: boolean;
  readonly reasonCode: "spam" | "harassment" | "impersonation" | "malicious_content" | "privacy_violation" | "unsafe_automation" | "other";
  readonly schemaVersion: "1.0";
  readonly targetEventId: string;
};

export type Provenance = "human" | "human_confirmed_agent" | "autonomous_agent";
